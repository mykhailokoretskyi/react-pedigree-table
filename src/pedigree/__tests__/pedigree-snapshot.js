import React from 'react';
import Pedigree from '../pedigree';
import renderer from '../../../node_modules/react/lib/ReactTestRenderer';

var instance = {
    father: {
        father: {
            father: {
                father: {
                    father: {},
                    mother: {}
                },
                mother: {
                    father: {},
                    mother: {}
                }
            },
            mother: {
                father: {
                    father: null,
                    mother: null
                },
                mother: {
                    father: null,
                    mother: null
                }
            }
        },
        mother: {
            father: {
                father: {
                    father: null,
                    mother: null
                },
                mother: {
                    father: null,
                    mother: null
                }
            },
            mother: {
                father: {
                    father: null,
                    mother: null
                },
                mother: {
                    father: null,
                    mother: null
                }
            }
        }
    },
    mother: {
        father: {
            father: {
                father: {
                    father: null,
                    mother: null
                },
                mother: {
                    father: null,
                    mother: null
                }
            },
            mother: {
                father: {
                    father: null,
                    mother: null
                },
                mother: {
                    father: null,
                    mother: null
                }
            }
        },
        mother: {
            father: null,
            mother: {
                father: {
                    father: null,
                    mother: null
                },
                mother: null
            }
        }
    }
};

describe('Pedigree', () => {
    it('matches snapshot with default generator', () => {
        const component = renderer.create(
            <Pedigree pedigreeData={instance}
                      leftBranchFieldName={"father"}
                      rightBranchFieldName={"mother"}
                      depth={5} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('matches snapshot with custom', () => {
        const generator = function(item){
            return (
                <div className="item">
                    <span>Real generator</span>
                </div>
            );
        };

        const component = renderer.create(
            <Pedigree pedigreeData={instance}
                      cellGenerator={generator}
                      leftBranchFieldName={"father"}
                      rightBranchFieldName={"mother"}
                      depth={5} />
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});