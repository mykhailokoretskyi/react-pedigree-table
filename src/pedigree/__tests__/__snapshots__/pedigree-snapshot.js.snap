exports[`Pedigree matches snapshot with custom 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          rowSpan={16}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={8}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={8}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={16}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={8}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={8}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="item">
            <span>
              Real generator
            </span>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Pedigree matches snapshot with default generator 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          rowSpan={16}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={8}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={8}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={16}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={8}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={8}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={4}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={2}>
          <div
            className="emptyCell" />
        </td>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
      <tr>
        <td
          rowSpan={1}>
          <div
            className="emptyCell" />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
